exercises/takudzwa_makoni/ex01_basics/src/exercise01.cpp

```diff
@@ -0,0 +1,100 @@
+
+#include "pch.h"
+#include <iostream>
+#include <fstream>
+#include <list>
+#include <iomanip>
+#include <string>
+#include <algorithm>
+#include <cctype>
+#include <iomanip>
+
+
+using namespace std;
+list<string> searchList = {"which","holmes","there","could","photograph"}; // list of words to be searched.
+
+
+// return true if special character exists in c.
+bool isSpecialCharacter(char c)
+{
+	switch (c)
+	{
+	case '(':
+	case ')':
+	case '.':
+	case ',':
+	case '"': // still need backslash?
+	case '?':
+	case '!':
+	case '\'':
+	case ':':
+		return true;
+	default:
+		return false;
+	}
+}
+
+
+//count number of occurences in file and cout result
+void printCount(const string& searchWord, ifstream& fileObj) {
+	
+	//std::string temp = searchWord; no longer required?
+
+	
+	int count = 0;
+	string word; //word from file
+	while (fileObj >> word) {
+		std::transform(word.begin(), word.end(), word.begin(), std::tolower); //convert all characters of word to lower
+		word.erase(std::remove_if(word.begin(), word.end(), &isSpecialCharacter), word.end()); //remove special characters from word.
```

Comment:
```suggestion
		std::remove_if(word.begin(), word.end(), &isSpecialCharacter);
```

Would be sufficient here.

Reply:
have made these changes